### PR TITLE
Allow to input new data in predict

### PR DESCRIPTION
**Pull request by pkofod**
This allows users to validate their fit on a hold-out data set or new data obtained later. Very handy to assess external validation.

### CODE REVIEW

File: src/estimation/diagnostics.jl [hunk 1]
```diff
@@ -516,24 +516,33 @@ function StatsBase.predict(model::PumasModel, subject::Subject, param, approx, v
   SubjectPrediction(pred, ipred, subject, approx)
 end
 
-function StatsBase.predict(fpm::FittedPumasModel, approx=fpm.approx; nsim=nothing, timegrid=false, newdata=false, useEBEs=true)
+function StatsBase.predict(fpm::FittedPumasModel, approx=fpm.approx; nsim=nothing, timegrid=false, data=nothing, useEBEs=true)
```
Comment (andreasnoack) [inline src/estimation/diagnostics.jl]: Did you consider doing this with dispatch?

Comment (pkofod) [reply src/estimation/diagnostics.jl]: I did, yes, the ifelse mess made me think to either have predict dispatch or at least change the internal part.

Comment (pkofod) [reply src/estimation/diagnostics.jl]: I’ll make variants with fpm and then a population (defaulting to the one in fpm) and then approx. They then map to get subject dispatch at the lowest level

**Comment by pkofod** [comment]
I think this is good to go, but let's wait for the master merge to finish its tests.

**Pull request merged**